devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -276,23 +278,27 @@ def parse(String description) {
 				}
 			} else if (descMap.attrInt == ATTRIBUTE_HEAT_SETPOINT) {
 				def intVal = Integer.parseInt(descMap.value, 16)
-				if (intVal != 0x8000) {		// 0x8000
+				if (intVal != 0x8000) {
 					log.debug "HEATING SETPOINT"
 					map.name = "heatingSetpoint"
 					map.value = getTemperature(descMap.value)
+					map.unit = getTemperatureScale()
 					map.data = [heatingSetpointRange: heatingSetpointRange]
 				}
 			} else if (descMap.attrInt == ATTRIBUTE_PI_HEATING_STATE) {
 				def intVal = Integer.parseInt(descMap.value, 16)
 				log.debug "HEAT DEMAND"
 				map.name = "thermostatOperatingState"
-				if (intVal < 10) {
+				if (intVal < 0x10) {
```

Comment:
@stelpro Just to check, we are comparing the value with 16, correct? I changed the code you originally submitted which compared "10" with the hex string. I just want to make sure that I am reading that correctly. (I accidentally just removed the quotes and didn't add the "0x" part.)

Reply:
Awesome. I updated this back to 10 in a new commit. When I look into some of the deeper logic handling on setpoint changes I will check what events I am receiving. I *might* remove this check and improve the 30 seconds logic below to mitigate the "flapping" effect. I'm thinking something that pays attention to the reported temperature and the setpoint as well.